sdk/cmake/CMakeMacros.cmake

```diff
@@ -765,8 +765,8 @@ function(create_registry_hives)
     # BootCD setup system hive
     add_custom_command(
         OUTPUT ${CMAKE_BINARY_DIR}/boot/bootdata/SETUPREG.HIV
-        COMMAND native-mkhive -h:SETUPREG -u -d:${CMAKE_BINARY_DIR}/boot/bootdata ${CMAKE_BINARY_DIR}/boot/bootdata/hivesys_utf16.inf ${CMAKE_SOURCE_DIR}/boot/bootdata/setupreg.inf
-        DEPENDS native-mkhive ${CMAKE_BINARY_DIR}/boot/bootdata/hivesys_utf16.inf)
+        COMMAND native-mkhive -h:SETUPREG -u -d:${CMAKE_BINARY_DIR}/boot/bootdata ${_registry_inf}
+        DEPENDS native-mkhive ${_registry_inf})
```

Comment:
That's a hack; 1st-stage should need just the hivesys contents + the setupreg.inf layout.

Reply:
1st stage needs critical device database, no matter how it gets there :)

Reply:
Technically the critical DB at 1st-stage comes from txtsetup.sif. But if you want it to come from the registry as a simplification, just remove the stuff in the setupreg.inf that precisely told the hive generator to remove the critical DB contents. This should make your patch actually better :P